bin/cask

```diff
@@ -236,26 +236,48 @@ def find_best_emacs():
     else:
         return 'emacs'
 
+def inside_emacs_24():
+    """Whether Cask is running inside Emacs-24.
+
+Emacs 24 is rather inconsistent in its use of the ``$EMACS`` and
+``$INSIDE_EMACS`` environment variables, so we try here to test if Cask is
+being run inside Emacs (either in a shell or with ``M-x compile``).
+
+Return ``True`` when running in Emacs 24, ``False`` otherwise.
+    """
+    ## are we inside Emacs at all
+    return (b'INSIDE_EMACS' in ENVB and
+            ## M-x compile in Emacs-24 sets INSIDE_EMACS to exactly 't'
+            (ENVB.get(b'INSIDE_EMACS') == 't'
+             or
```

Comment:
Uhm, it does, doesn't it?

![bildschirmfoto 2016-03-18 um 18 27 50](https://cloud.githubusercontent.com/assets/224922/13886453/2ca02d90-ed37-11e5-8f92-973179ba8e94.png)